rgw/v2/tests/s3_swift/test_quota_management.py

```diff
@@ -0,0 +1,195 @@
+"""test_quota_management - Test quota management
+
+Usage: test_quota_management.py -c <input_yaml>
+
+<input_yaml>
+        Note: any one of these yamls can be used
+        test_quota_management.yaml
+
+Operation:
+    Create non tenanted user
+    Create bucket
+    set bucket quota
+    enable bucket quota
+    test bucket quota
+    disable bucket quota
+    set user quota
+    enable user quota
+    test user quota
+    disable user quota
+"""
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import argparse
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.resource_op import Config
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+
+TEST_DATA_PATH = None
+import logging
+
+log = logging.getLogger()
+
+
+# create tenanted and non tenanted user
+# create buckets for both users
+# rename buckets and users
+
+
+def upload_object_initiate(
+    config, user_info, bucket, bucket_name, obj_name_suffix, obj_size
+):
+    config.obj_size = obj_size
+    s3_object_name = utils.gen_s3_object_name(bucket_name, obj_name_suffix)
+    log.info("s3 object name: %s" % s3_object_name)
+    s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+    log.info("s3 object path: %s" % s3_object_path)
+    log.info("upload type: normal")
+    reusable.upload_object(
+        s3_object_name,
+        bucket,
+        TEST_DATA_PATH,
+        config,
+        user_info,
+    )
+
+
+def test_exec(config):
+
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+
+    log.info("Creating an user")
+    config.user_count = 1
+    user_info = s3lib.create_users(config.user_count)
+    user_info = user_info[0]
+    auth = Auth(user_info, ssl=config.ssl)
+    rgw_conn = auth.do_auth()
+
+    log.info("Creating a bucket")
+    bucket_name = utils.gen_bucket_name_from_userid(user_info["user_id"], rand_no=1)
+    bucket = reusable.create_bucket(bucket_name, rgw_conn, user_info)
+
+    reusable.set_bucket_quota(
+        user_info, config.bucket_max_objects, config.bucket_max_size
+    )
+    reusable.enable_bucket_quota(user_info)
+
+    try:
+        upload_object_initiate(
+            config, user_info, bucket, bucket_name, 1, config.bucket_max_size + 100
+        )
+    except TestExecError as e:
+        log.info("bucket quota with max size passed")
+    except Exception:
```

Comment:
We can have
except:
        raise AssertionError("bucket quota with max size failed")
  Then we dont have to provide sys.exit(1) and we can skip else part